src/content-proxy.js

```diff
@@ -47,20 +63,27 @@ async function contentProxy(opts) {
   url.searchParams.append('rid', options.requestId);
 
   log.info(`Fetching content from: ${url.href}`);
-  const response = await fetch(url.href, getFetchOptions(options));
-  const body = await response.text();
-  if (response.ok) {
+  const resp = await fetch(url.href, getFetchOptions(options));
+  const body = await resp.buffer();
+  if (resp.ok) {
+    const headers = {};
+    PASSTHROUGH_HEADERS.forEach((name) => {
```

Comment:
This would be better as a reduce.

Reply:
OK, but if I change it to:
```
    const headers = PASSTHROUGH_HEADERS.reduce((previous, name) => {
      const value = resp.headers.get(name);
      if (value) {
        previous[name] = value;
      }
      return previous;
    }, {});
```
I have to disable `eslint(no-param-reassign)` which is probably why I avoid `reduce`.

Reply:
I also don't like reduce either for these kind of operations....